tests/test_workarounds.py

```diff
@@ -56,3 +61,49 @@ def test_build_string_filters():
             assert pkg["version"].startswith("3.8")
         if pkg["name"] == "numpy":
             assert "py38" in pkg["build_string"]
+
+
+@pytest.mark.parametrize("stage", ["Collecting package metadata", "Solving environment"])
+def test_ctrl_c(stage):
+    p = sp.Popen(
+        [
+            sys.executable,
+            "-m",
+            "conda",
+            "create",
+            "-p",
+            "UNUSED",
+            "--dry-run",
+            "--solver=libmamba",
+            "--override-channels",
+            "--channel=conda-forge",
+            "vaex",
+        ],
+        text=True,
+        stdout=sp.PIPE,
+        stderr=sp.PIPE,
+    )
+    t0 = time.time()
+    while stage not in p.stdout.readline():
+        time.sleep(0.1)
+        if time.time() - t0 > 30:
+            raise RuntimeError("Timeout")
+
+    # works around Windows' awkward CTRL-C signal handling
+    # https://stackoverflow.com/a/64357453
+    if sys.platform == "win32":
+        kernel = ctypes.windll.kernel32
+        kernel.FreeConsole()
+        kernel.AttachConsole(p.pid)
+        kernel.SetConsoleCtrlHandler(None, 1)
+        kernel.GenerateConsoleCtrlEvent(0, 0)
+        p.wait(timeout=30)
+        assert p.returncode != 0
+        assert "KeyboardInterrupt" in p.stdout.read() + p.stderr.read()
+        kernel.SetConsoleCtrlHandler(None, 0)
```

Comment:
Should this line be added earlier via a `request.addfinalizer()`, in case the assertions do not work (e.g. connection error). Or a `try / finally` block.

Reply:
yeah, lemme add this